keras_cv/losses/iou_loss.py

```
@@ -0,0 +1,93 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+
+
+class IoULoss(tf.keras.losses.Loss):
+    """Implements the IoU Loss
+
+    IoU loss is commonly used for object detection. This loss aims to directly
+    optimize the IoU score between true boxes and predicted boxes. The length of the
+    last dimension should be atleast 4 to represent the bounding boxes.
+
+    Args:
+        bounding_box_format: a case-insensitive string which is one of `"xyxy"`,
+            `"rel_xyxy"`, `"xyWH"`, `"center_xyWH"`, `"yxyx"`, `"rel_yxyx"`. The
+            position and shape of the bounding box will be followed by the class and
+            confidence values (in that order). This is required for proper ranking of
+            the bounding boxes. Therefore, each bounding box is defined by 6 values.
+            For detailed information on the supported format, see the
+            [KerasCV bounding box documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
+        mode: must be one of
+            - `"linear"`. The loss will be calculated as 1 - iou
+            - `"squared"`. The loss will be calculated as 1 - iou<sup>2</sup>
+            - `"log"`. The loss will be calculated as -ln(iou)
+            Defaults to "log".
+
+    References:
+        - [UnitBox paper](https://arxiv.org/pdf/1608.01471)
+
+    Sample Usage:
+    ```python
+    y_true = tf.random.uniform([5, 10, 6], minval=0, maxval=10, dtype=tf.dtypes.int32)
+    y_pred = tf.random.uniform([5, 10, 4], minval=0, maxval=10, dtype=tf.dtypes.int32)
+    loss = IoULoss(bounding_box_format = "xyWH")
+    loss(y_true, y_pred).numpy()
+    ```
+    Usage with the `compile()` API:
+    ```python
+    model.compile(optimizer='adam', loss=keras_cv.losses.IoULoss())
+    ```
+    """
+
+    def __init__(self, bounding_box_format, mode="log", **kwargs):
```

Do you anticipate both inputs will be from different formats?

---

Yeah, we need to support all bounding box formats.  This is critical to our API consistency, and our continued effort to minimize silent errors.

---

Does that mean I should add two `bounding_box_format` variables to the constructor, i.e. one for true and one for pred?

---

No, It’s fine as is feel free to resolve